Add UserAssignmentAPI and update UserOccurrenceAPI

TimvdLippe opened the pull request.

TimvdLippe committed:
Add initial version of UserAssignmentAPI

jwgmeligmeyling commented on src/main/java/nl/tudelft/planningstool/api/v1/ResponseAPI.java:
```diff
@@ -0,0 +1,31 @@
+    @Inject
+    protected CourseDAO courseDAO;
+
+    protected <T, R> ListResponse<R> createListResponse(Collection<T> occurrences, Function<T, R> mapper) {
```
`Function<? super T, R>`

jwgmeligmeyling commented on src/main/java/nl/tudelft/planningstool/api/v1/UserAssignmentAPI.java:
```diff
@@ -0,0 +1,38 @@
+        return createListResponse(assignments);
+    }
+
+    private ListResponse<AssignmentResponse> createListResponse(Collection<Assignment> assignments) {
```
`private ListResponse<AssignmentResponse> createListResponse(Collection<? extends Assignment> assignments)`

TimvdLippe committed:
Update to new occurrenceResponse format

TimvdLippe committed:
Change time formats in occurrences for client.

TimvdLippe committed:
Correct timestamps of events and enabled -> editable.

TimvdLippe committed:
Add filtering on timeslot based on query parameters start and end

TimvdLippe committed:
Clean up API

- Remove unused MainAPI
- Add timeslots for the json feed
- Fix json feed to return a collection instead of ListResponse (for fullcalendar.js)
- Add AppTest to verify Guice is correctly configured

TimvdLippe committed:
Fix TimeZone issue with TimeSlot

TimvdLippe committed:
Make TimeSlot boundaries inclusive for occurrence fetching

TimvdLippe committed:
Add endpoint for creation of occurrences

TimvdLippe committed:
Update occurrence creation format according to conversation

TimvdLippe committed:
Fix occurrence response to include all subclass fields

TimvdLippe committed:
Add PUT method for UserOccurrences

TimvdLippe committed:
Update processing PUT requests of client

TimvdLippe committed:
Fix TimeZone conversions and editable field of OccurrenceResponse

TimvdLippe committed:
Add Occurrence check for user and merge the Occurrence to save it

TimvdLippe commented:
Thoroughly tested for first demo. Looks good.
